libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/utils/DirectionsRouteMissingConditionsCheckTest.kt

```diff
@@ -0,0 +1,106 @@
+package com.mapbox.navigation.base.internal.utils
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import junit.framework.Assert.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.net.URL
+
+class DirectionsRouteMissingConditionsCheckTest {
+
+    private val exclusiveQueries = mapOf(
+        "engine" to "electric",
+        "key1" to "value1",
+    )
+
+    @Before
+    fun setup() {
+        mockkObject(QueriesProvider)
+        every { QueriesProvider.provideQueries() } returns exclusiveQueries
+    }
+
+    @After
+    fun cleanUp() {
+        unmockkObject(QueriesProvider)
+    }
+
+    @Test
+    fun cases() {
```

Comment:
Don't you like using Parameterized tests? This way they are run as separate test so that if one case fails the following ones will still be executed. And the error message is more clear.

Not important for this particular PR tho, just curious.

Reply:
TBH, I find the separate tests more readable than parameterized. Like
```kotlin
@Test
fun `Not restricted key-value pair`() {
    val route = createDirectionsRouteWithUrl("https://test.com/?key0=value0")
    DirectionsRouteMissingConditionsCheck.checkDirectionsRoute(route)
}
```

Reply:
all 3 variants are great, let's not focus on the approach 🥲 I added all cases in the single function just because it was faster to add